src/backend/Coq.ml

```diff
@@ -44,55 +45,55 @@ module Make(S : Msat.S)(A : Arg with type hyp := S.clause
     Array.iter aux (Clause.atoms clause)
 
   let elim_duplicate fmt goal hyp _ =
-    (** Printing info comment in coq *)
-    Format.fprintf fmt
-      "(* Eliminating doublons. Goal : %s ; Hyp : %s *)@\n"
+    (* Printing info comment in coq *)
+    Format.fprintf fmt "(* Eliminating doublons. Goal : %s ; Hyp : %s *)@\n"
       (name goal) (name hyp);
-    (** Prove the goal: intro the atoms, then use them with the hyp *)
+    (* Prove the goal: intro the atoms, then use them with the hyp *)
     let m = clause_map goal in
     Format.fprintf fmt "pose proof @[<hov>(fun %a=>@ %s%a) as %s@].@\n"
-      (clause_iter m "%s@ ") goal (name hyp)
-      (clause_iter m "@ %s") hyp (name goal)
+      (clause_iter m "%s@ ") goal (name hyp) (clause_iter m "@ %s") hyp
+      (name goal)
 
   let resolution_aux m a h1 h2 fmt () =
     Format.fprintf fmt "%s%a" (name h1)
-      (fun fmt -> Array.iter (fun b ->
-           if b == a then begin
-             Format.fprintf fmt "@ (fun p =>@ %s%a)"
-               (name h2) (fun fmt -> (Array.iter (fun c ->
-                   if Atom.equal c (Atom.neg a) then
-                     Format.fprintf fmt "@ (fun np => np p)"
-                   else
-                     Format.fprintf fmt "@ %s" (M.find c m)))
-                 ) (Clause.atoms h2)
-           end else
-             Format.fprintf fmt "@ %s" (M.find b m)
-         )) (Clause.atoms h1)
+      (fun fmt ->
+        Array.iter (fun b ->
+            if b == a then
+              Format.fprintf fmt "@ (fun p =>@ %s%a)" (name h2)
+                (fun fmt ->
+                  Array.iter (fun c ->
+                      if Atom.equal c (Atom.neg a) then
+                        Format.fprintf fmt "@ (fun np => np p)"
+                      else
+                        Format.fprintf fmt "@ %s" (M.find c m) ) )
+                (Clause.atoms h2)
+            else
+              Format.fprintf fmt "@ %s" (M.find b m) ) )
+      (Clause.atoms h1)
 
   let resolution fmt goal hyp1 hyp2 atom =
     let a = Atom.abs atom in
     let h1, h2 =
-      if Array.exists (Atom.equal a) (Clause.atoms hyp1) then hyp1, hyp2
+      if Array.exists (Atom.equal a) (Clause.atoms hyp1) then
+        hyp1, hyp2
       else (
         assert (Array.exists (Atom.equal a) (Clause.atoms hyp2));
-        hyp2, hyp1
-      )
+        hyp2, hyp1 )
```

Comment:
In if-then-elses that need parentheses, since the starting parenthesis is not on the same line as the start of the expression, is it possible to have the ending parenthesis on its line ?

Reply:
no, I looked and didn't find anything, sadly. It's one of the things I dislike too. But it still has the advantage of consistency.

Reply:
Indeed, and I can guess I can live with it (though it would be interesting to ask the ocamlformat devs if it's possible to add an option for that, I'll look at it when I return from holidays).